### DISCUSSION TITLE
Is this project going to be maintained or is it a temporary thought experiment?

### DISCUSSION THREAD

**Discussion by ibehnam** [guidance-ai/guidance#201] [General]
I appreciate your effort in making guidance. Really interesting ideas especially for low-level control of LLMs. But something that makes me worried about using guidance in my projects is the uncertainty about its future. With Langchain, we know that the community and developers are going to keep improving it. But can we expect the same from guidance? I know the authors are prolific (e.g. @slundberg is known for SHAP, a really useful XAI method that lots of people use nowadays). Obviously, I'm not asking for a guarantee or assurance from the authors since this is an open source project and AFAIK it's not funded or anything. I just wanted to see if guidance will be a good choice for long-term projects and whether there are plans to keep improving it as new LLMs are introduced?

**Comment by ibehnam** [comment]
I guess the lack of response by the project creators answers my question.

**Comment by faizanahemad** [comment]
Would be great to get author comments here. @slundberg

**Comment by rickbraddy-pharma** [comment]
Growing increasingly concerned about the lack of progress on serious bugs introduced into recent releases and lack of visible commitment or traction to resolve them.

Looks like Microsoft needs to step up with resources to address this before it’s too late…

**Comment by ibehnam** [comment]
I mean it's already too late. I stopped using Guidance a while ago. Turns out, the number of stars on GH doesn't mean much. If a project is dead, it's time to move on.

**Reply by ein-ich** [reply]
Doesn't seem to be dead, see branch pythonic.

**Reply by XBeg9** [reply]
Can you expose, what are you using now instead of guidance?

**Reply by atcol** [reply]
@ibehnam what are you using instead of Guidance, please?

**Reply by ibehnam** [reply]
@atcol LMQL.

**Comment by rickbraddy-pharma** [comment]
We aren't giving up yet.... we will start looking to help with fixes next week... 
But, we DO need to hear from the project's Leadership on the plan, direction and commitment (and whether this is something Microsoft cares about supporting moving forward).

**Comment by atcol** [comment]
I have discovered this project recently and have used it as the next iteration in some internal tooling at my workplace. I'd very much like to continue to use it but I am unsure as to whether it will be developed further. 

Any comment from the author(s)/maintainer(s) would be very much appreciated. Thank you!

**Comment by ibehnam** [comment]
I found LMQL to be much more actively developed and with more features.

**Comment by faizanahemad** [comment]
This project seems dead by the commit history in `main` and in `pythonic` branch as well. In time that this project gets one commit, langchain gets a new version. I would be fine if this was Apache server or some other well polished software but low traction at such an early stage = dead project.

**Reply by ein-ich** [reply]
Sad but I'll stop using it as well, I'm still using an even older version because of "newly" introduced bugs.

**Comment by emrekiciman** [comment]
FYI see #337 Guidance Looking Forward

**Comment by faizanahemad** [comment]
Is this maintained or support is removed for any future works?
